mlir/lib/Transforms/Utils/LoopUtils.cpp

```diff
@@ -1653,6 +1653,24 @@ void mlir::scalarReplace(AffineForOp forOp) {
   LLVM_DEBUG(llvm::dbgs() << "SCAL REP END\n");
 }
 
+void mlir::affineParallelize(AffineForOp forOp) {
+  OpBuilder outsideBuilder(forOp);
+  Location loc = forOp.getLoc();
+  BlockAndValueMapping operandMap;
+  auto forOpIV = forOp.getInductionVar();
+  auto newPloop = outsideBuilder.create<AffineParallelOp>(
+      loc, forOp.getLowerBoundMap(), forOp.getLowerBoundOperands(),
+      forOp.getUpperBoundMap(), forOp.getUpperBoundOperands());
+  auto parallelIV = newPloop.getIVs()[0];
+  operandMap.map(forOpIV, parallelIV);
+  OpBuilder ob(newPloop);
+  ob.setInsertionPointToStart(newPloop.getBody());
+  for (auto &op : forOp.getBody()->without_terminator()){
+    ob.clone(op, operandMap);
+  }
```

Comment:
Please add comments at appropriate places for readability. For eg: "// Replace affine.for with a 1-d affine.parallel and clone the former's body into the latter while remapping values.".